src/Trace/TraceSpan.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+use Google\Cloud\Core\ArrayTrait;
+
+/**
+ * This plain PHP class represents a
+ * [TraceSpan resource](https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#TraceSpan)
+ * A span represents a single timed event within a Trace. Spans can be nested
+ * and form a trace tree. Often, a trace contains a root span that describes
+ * the end-to-end latency of an operation and, optionally, one or more subspans
+ * for its suboperations. Spans do not need to be contiguous. There may be
+ * gaps between spans in a trace.
+ */
+class TraceSpan
+{
+    use ArrayTrait;
+
+    const SPAN_KIND_UNSPECIFIED = 'SPAN_KIND_UNSPECIFIED';
+    const SPAN_KIND_RPC_SERVER = 'RPC_SERVER';
+    const SPAN_KIND_RPC_CLIENT = 'RPC_CLIENT';
+
+    /**
+     * Associative array containing all the fields representing this TraceSpan.
+     * @var array
+     */
+    private $info;
+
+    /**
+     * Instantiate a new TraceSpan instance.
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $spanId The ID of the span. If not provided,
+     *            one will be generated automatically for you.
+     *      @type string $kind Distinguishes between spans generated
+     *            in a particular context. **Defaults to**
+     *            SPAN_KIND_UNSPECIFIED.
+     *      @type string $name The name of the span.
+     *      @type string $startTime Start time of the span in
+     *            nanoseconds in "Zulu" format.
+     *      @type string $endTime End time of the span in
+     *            nanoseconds in "Zulu" format.
+     *      @type string $parentSpanId ID of the parent span if any.
+     *      @type array $labels Associative array of $label => $value
+     *            to attach to this span.
+     * }
+     */
+    public function __construct($options = [])
+    {
+        $this->info = $this->pluckArray(
+            ['spanId', 'kind', 'name', 'startTime', 'endTime', 'parentSpanId', 'labels'],
+            $options
+        );
+        $this->info += [
+            'kind' => self::SPAN_KIND_UNSPECIFIED
+        ];
+
+        if (!array_key_exists('spanId', $this->info)) {
```

Comment:
I didn't want to run those generators unless the key was missing. They're not doing a ton, but it still seemed worthwhile to skip.